### DISCUSSION TITLE
Version without argon2 possible?

### DISCUSSION THREAD

**Discussion by alexzaw4956** [coder/code-server#7055] [Ideas]
First of all, 
I want to say thank you guys very much for giving us the ability to code anywhere. 
I have been trying to code-server installed on an IBM i machine OS400 PPC64 architecture. Yes, I know we are a forgotten bunch, but there are a lot of us who utilize one of the extensions from VSCode called code-for-i, and it helps us get away from something called "green screen".

Long story short, 
I was never able to get it installed due to one thing or another, after trying so many things the last hurdle was argon2, it wouldn't install on IBM i, even after trying so many different paths.... It is one year later and I am a year wiser, so decided to give it a try again, this time I was able to scan the mention of argon2 in all the js files, and good thing it was only in utils.js that mattered. After making some adjustments to use just crypto (I know.. I know.. less secure) and ./bin/code-server and voila!!

So, my hurdle had been solved, and I am sure there are others, as I had read on my different forums during my one year journey, can benefit from this.

**Comment by code-asher** [comment]
Good question, I think it would be possible to make argon2 an optional dependency by importing it dynamically when an argon2-style `hashed-password` is being used.
